sdcm/sct_config.py

```diff
@@ -2291,6 +2297,62 @@ def verify_configuration_urls_validity(self):  # pylint: disable=too-many-branch
         get_branch_version_for_multiple_repositories(
             urls=(self.get(url) for url in repos_to_validate if self.get(url)))
 
+    def get_version_based_on_conf(self):  # pylint: disable=too-many-locals
+        """
+        figure out which version and if it's enterprise version
+        base on configuration only, before nodes are up and running
+        so test configuration can set up things which need to happen
+        before nodes are up
+
+        this is information is cached on the SCTConfiguration object
+        :return: tuple - (scylla_version, is_enterprise)
+        """
+        backend = self.get('cluster_backend')
+        scylla_version = None
+        _is_enterprise = False
+
+        if not self.get('use_preinstalled_scylla'):
+            scylla_repo = self.get('scylla_repo')
+            scylla_version = get_branch_version(scylla_repo)
+            _is_enterprise = is_enterprise(scylla_version)
+        elif unified_package := self.get('unified_package'):
+            with tempfile.TemporaryDirectory() as tmpdirname:
+                LOCALRUNNER.run(shell_script_cmd(f"""
+                    cd {tmpdirname}
+                    curl {unified_package} -o ./unified_package.tar.gz
+                    tar xvfz ./unified_package.tar.gz
+                    """), verbose=False)
+
+                scylla_version = next(pathlib.Path(tmpdirname).glob('**/SCYLLA-VERSION-FILE')).read_text()
+                scylla_product = next(pathlib.Path(tmpdirname).glob('**/SCYLLA-PRODUCT-FILE')).read_text()
+                _is_enterprise = scylla_product == 'scylla-enterprise'
+
+        elif self.get('db_type') == 'cloud_scylla':
+            _is_enterpise = True
+        elif backend == 'aws':
+            amis = self.get('ami_id_db_scylla').split()
+            region_name = self.region_names[0]
+            tags = get_ami_tags(ami_id=amis[0], region_name=region_name)
+            scylla_version = tags.get('scylla_version')
+            _is_enterprise = is_enterprise(scylla_version)
+        elif backend == 'gce':
+            images = self.get('gce_image_db').split()
+            tags = get_gce_image_tags(images[0])
+            scylla_version = tags.get('scylla_version').replace('-', '.')
+            _is_enterprise = is_enterprise(scylla_version)
+        elif backend == 'azure':
+            images = self.get('azure_image_db').split()
+            tags = azure_utils.get_image_tags(images[0])
+            scylla_version = tags.get('scylla_version')
+            _is_enterprise = is_enterprise(scylla_version)
+        elif backend == 'docker' or 'k8s' in backend:
+            docker_repo = self.get('docker_image')
+            scylla_version = self.get('scylla_version')
+            _is_enterprise = 'enterprise' in docker_repo
+        self.scylla_version = scylla_version
+        self.is_enterprise = _is_enterprise
+        return scylla_version, _is_enterprise
```

Comment:
What about the actual scylla_version that we get from scylla?
Do we want to compare them at one point to make sure we figured it out correctly from the configuration?
And what eventually we will update to Argus and report, the one from conf or the one we had till now?

Reply:
we could, I'm not sure it would help us that much anything.

this PR doesn't change how we report to Argus. (we could add this version as well, but it's not very critical)